File: gitdot-web/packages/api/src/endpoint/review/list-reviews.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListReviewsRequest = z.object({});
export type ListReviewsRequest = z.infer<typeof ListReviewsRequest>;

export const ListReviewsResponse = z.array(ReviewResource);
export type ListReviewsResponse = z.infer<typeof ListReviewsResponse>;

export const ListReviews = {
  path: "/repository/{owner}/{repo}/reviews",
  method: "GET",
  request: ListReviewsRequest,
  response: ListReviewsResponse,
} as const satisfies Endpoint;
export type ListReviews = typeof ListReviews;

File: gitdot-api/src/endpoint/review/list_reviews.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewResource};

pub struct ListReviews;

impl Endpoint for ListReviews {
    const PATH: &'static str = "/repository/{owner}/{repo}/reviews";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListReviewsRequest;
    type Response = ListReviewsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct ListReviewsRequest;

pub type ListReviewsResponse = Vec<ReviewResource>;
